src/LayoutSaver.cpp

```diff
@@ -339,10 +340,15 @@ void LayoutSaver::Private::deserializeWindowGeometry(const T &saved, QWidgetOrQu
     // Not simply calling QWidget::setGeometry() here.
     // For QtQuick we need to modify the QWindow's geometry.
 
+    QRect geometry = saved.geometry;
+    if (!isNormalWindowState(saved.windowState)) {
```

Comment:
typo ? if it's maximized it will use the normal size ?

Reply:
Not typo
Normal geometry required for restoring geometry of windows with non-normal state 
After setting normal geometry we will set a state, and a window will maximized